jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/JaxRsResourceBase.java

```diff
@@ -533,4 +535,48 @@ protected void verifyNumberOfElements(int actual, int expected, String errorMess
         Preconditions.checkArgument(actual == expected, errorMessage);
     }
 
+    protected Response createPaymentResponse(final UriInfo uriInfo, final Payment payment, final TransactionType transactionType, @Nullable final String transactionExternalKey) {
+        final PaymentTransaction createdTransaction = findCreatedTransaction(payment, transactionType, transactionExternalKey);
+        Preconditions.checkNotNull(createdTransaction, "No transaction of type '%s' found", transactionType);
```

Comment:
> we don't ever expect aborted payments right now

Well, who is `we`? 😸  It's up to plugin developers to abort payments in their control plugin if needed (e.g. [fraud detection systems](https://github.com/killbill/killbill-accertify-plugin/blob/master/src/main/java/org/killbill/billing/plugin/accertify/api/AccertifyPaymentControlPluginApi.java#L71-L81)).